### PR TITLE
KOGITO-2237: Stunner - Domain Model Improvements

### PR DESCRIPTION
**Pull request by romartin**
See https://issues.redhat.com/browse/KOGITO-2237

### CODE REVIEW

**Comment by romartin** [comment]
Jenkins execute full downstream build

File: kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/definition/adapter/binding/ClientBindingUtils.java [hunk 1]
```diff
@@ -85,30 +110,7 @@
...
-                            LOGGER.log(Level.INFO,
-                                       "BindableAdapterUtils#merge - Skipping merge property [" + pId + "]");
-                        }
-                    }
-                    return (R) target;
-                }
-            }
-        }
-        return null;
+    private static boolean isNotEmpty(final String s) {
```
Comment (hasys) [inline kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/definition/adapter/binding/ClientBindingUtils.java:113]: You can use `org.kie.workbench.common.stunner.core.util.StringUtils#nonEmpty`

File: kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-core-common/src/main/java/org/kie/workbench/common/stunner/core/definition/clone/DeepCloneProcess.java [hunk 2]
```diff
@@ -50,29 +52,24 @@ public DeepCloneProcess(final FactoryManager factoryManager,
...
-                            .findFirst();
-                    return propertyTarget.isPresent() ? new AbstractMap.SimpleEntry(p,
-                                                                                    propertyTarget.get()) : null;
-                })
-                .filter(Objects::nonNull)
-                .filter(entry -> isAllowedToClone(adapterManager.forProperty().getValue(entry.getKey())))
-                .forEach(entry -> {
-                    Object value = adapterManager.forProperty().getValue(entry.getKey());
-                    adapterManager.forProperty().setValue(entry.getValue(),
-                                                          cloneValue(value));
```
Comment (hasys) [inline kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-core-common/src/main/java/org/kie/workbench/common/stunner/core/definition/clone/DeepCloneProcess.java:73]: `cloneValue` private method is not used any more and can be deleted.

File: kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-api/kie-wb-common-stunner-core-api/src/main/java/org/kie/workbench/common/stunner/core/definition/adapter/binding/DefinitionAdapterBindings.java [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+
+        public void setWidthIndex(int index) {
+            metaTypeFieldIndexes[WIDTH] = index;
+        }
+
+        public void setHeightIndex(int index) {
+            metaTypeFieldIndexes[HEIGHT] = index;
+        }
+
+        public void setRadiustIndex(int index) {
```
Comment (hasys) [inline kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-api/kie-wb-common-stunner-core-api/src/main/java/org/kie/workbench/common/stunner/core/definition/adapter/binding/DefinitionAdapterBindings.java:210]: Typo in method name

**Comment by romartin** [comment]
Hey @hasys - I've jsut rebased and fixed your review comments, thanks!

Hey @karreiro - added u also as a review for the DMN part, is it ok? This PR is about internal updates only, so everything should work as before

Thanks guys!

**Comment by romartin** [comment]
Jenkins execute full downstream build

**Comment by karreiro** [comment]
Jenkins execute full downstream build

**Comment by karreiro** [comment]
Jenkins execute full downstream build.

**Comment by hasys** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
Rebased guys, let's see the build now
Thanks!

**Comment by romartin** [comment]
Jenkins execute full downstream build

**Comment by hasys** [comment]
Jenkins execute compile downstream build

**Comment by karreiro** [comment]
Jenkins execute compile downstream build.

**Comment by romartin** [comment]
Jenkins execute compile downstream build

**Comment by romartin** [comment]
rebased again @karreiro , let's see if we can get some success build ... :/

**Comment by romartin** [comment]
Jenkins execute compile downstream build

**Comment by romartin** [comment]
Jenkins execute full downstream build

**Comment by karreiro** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
ooops @karreiro , honestly I'm just testing the `dmn-webapp-kogito-runtime`... and it's working perfectly! So sorry if any inconveniences, lemme test other webapps and let you know in a while.
Thanks for pointing this out!!!

**Comment by romartin** [comment]
oh thanks a lot @karreiro , great catch! I've been testing and the webapps which were using backend services were broken. I've added the fix. I've tested the DMN showcases and are working properly.

Thanks a lot again for this and sorry for the inconveniences. Fix pushed and ready for review again!! :+1:

**Comment by romartin** [comment]
Jenkins execute compile downstream build

**Comment by romartin** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
Jenkins execute compile downstream build

**Comment by romartin** [comment]
Jenkins execute compile downstream build

**Comment by romartin** [comment]
Hey @karreiro 

Can you please provide some feedback on this? I would like trying to merge before going to vacation. Thanks!

**Comment by romartin** [comment]
thanks @karreiro , good catch again! :+1: 
This looks not a regular "resize" operation, so not sure about what could be wrong yet, just lemme take a quick look on it... thx again! Will also try to do also more tests specially on DMN side, trying not to spend your time on iteration reviews.. sorry for this.

**Comment by romartin** [comment]
Jenkins execute compile downstream build

**Comment by romartin** [comment]
Hey @karreiro 

Thanks a lot for your help on this!! I've properly fixed the issue you mentioned, also did a deep look on code (based on this fix) and everything looks good. Also tested both standalone and kogito showcases for DMN and working properly, at least doing some tests.

Sooooo, please this should be ready for review again, sorry for bothering!!!

Thanks again!!

**Comment by romartin** [comment]
Hey @karreiro 

Thanks for the feedback again! I've properly fixed the issue in the way that no other webapps (like the drools-wb or jbpm-wb) will be affected. I've properly tested and working fine.

So I've  rebased, fixed and squashed all commits, everything is ready for review again. 

Thanks again for the collaboration here!

**Comment by romartin** [comment]
Jenkins execute fdb

**Comment by romartin** [comment]
Jenkins execute cdb

**Comment by romartin** [comment]
Hey @karreiro 

Thanks for the help! Anyway I couldn't reproduce the issue you mention in the `drools-wb`. I can properly open diagrams, create, save, etc. Any concrete step to reproduce? Are you using an empty project? Clean app?

I've rebased and force pushed, let's see if that fixes the issue you mention.

Thx!

**Comment by romartin** [comment]
Jenkins execute fdb

**Comment by romartin** [comment]
Jenkins execute cdb

**Comment by mbiarnes** [comment]
FDB test comment

**Comment by mbiarnes** [comment]
FDB test comment

**Comment by romartin** [comment]
Merging. I've tested the full bulid locally and it works.

**Pull request merged**